assets/test/spec/services/apiServiceSpec.js

```diff
@@ -0,0 +1,204 @@
+"use strict";
+
+describe("APIService", function(){
+  var APIService;
+
+  beforeEach(function(){
+    inject(function(_APIService_){
+      APIService = _APIService_;
+    });
+  });
+
+  describe("#toResourceGroupVersion", function(){
+
+    var tc = [
+      // string args
+      // simple
+      ['pods',         {r:'pods',g:'',v:'v1'}],
+      // normalization
+      ['Pods',         {r:'pods',g:'',v:'v1'}],
+      // normalization preserves subresources
+      ['PODS/FOO',     {r:'pods/FOO',g:'',v:'v1'}],
+      
+      // structured, resource only
+      // simple
+      [{resource:'pods'},         {r:'pods',g:'',v:'v1'}],
+      // normalization
+      [{resource:'Pods'},         {r:'pods',g:'',v:'v1'}],
+      // normalization preserves subresources
+      [{resource:'PODS/FOO'},     {r:'pods/FOO',g:'',v:'v1'}],
+
+      // structured, with group
+      // groups default version if known
+      [{resource:'pods',group:''},           {r:'pods',g:'',          v:'v1'     }],
+      [{resource:'jobs',group:'extensions'}, {r:'jobs',g:'extensions',v:'v1beta1'}],
+      // unknown groups do not default version
+      [{resource:'foos',group:'unknown'},    {r:'foos',g:'unknown',   v:undefined}],
+      
+      // structured, with version
+      // groups default
+      [{resource:'pods',version:'v1'},      {r:'pods',g:'',v:'v1'     }],
+      [{resource:'pods',version:'v1beta3'}, {r:'pods',g:'',v:'v1beta3'}],
+      
+      // structured, fully specified
+      [{resource:'pods',group:'',          version:'v1'},      {r:'pods',g:'',          v:'v1'     }],
+      [{resource:'pods',group:'',          version:'v1beta3'}, {r:'pods',g:'',          v:'v1beta3'}],
+      [{resource:'jobs',group:'extensions',version:'v1'},      {r:'jobs',g:'extensions',v:'v1'     }],
+      [{resource:'jobs',group:'extensions',version:'v1beta1'}, {r:'jobs',g:'extensions',v:'v1beta1'}],
+      [{resource:'foos',group:'unknown',   version:'v1'},      {r:'foos',g:'unknown',   v:'v1'     }],
+      [{resource:'foos',group:'unknown',   version:'v1beta1'}, {r:'foos',g:'unknown',   v:'v1beta1'}]
+    ];
+
+    angular.forEach(tc, function(item) {
```

Comment:
`_.spread` is nice for this, so you don't have to keep track of the array notation: 

```javascript
// can now name the items in the array as function args....
angular.forEach(tc, _.spread(function(toTest, expected) {

```